compiler/qsc_circuit/src/circuit.rs

```diff
@@ -0,0 +1,612 @@
+                Wire::Classical { start_column } => {
+                    if start_column.is_some() {
+                        self.add(column, CLASSICAL_WIRE_CROSS);
+                    } else {
+                        self.add(column, VERTICAL);
+                    }
+                }
+            }
+        }
+    }
+
+    fn add_dashed_vertical(&mut self, column: usize) {
+        if self.objects.get(&column).is_none() {
+            match self.wire {
+                Wire::Qubit { .. } => self.add(column, QUBIT_WIRE_DASHED_CROSS),
+                Wire::Classical { start_column } => {
+                    if start_column.is_some() {
+                        self.add(column, CLASSICAL_WIRE_DASHED_CROSS);
+                    } else {
+                        self.add(column, VERTICAL_DASHED);
+                    }
+                }
+            }
+        }
+    }
+
+    fn start_classical(&mut self, column: usize) {
+        self.add(column, CLASSICAL_WIRE_START);
+        if let Wire::Classical { start_column } = &mut self.wire {
+            start_column.replace(column);
+        }
+    }
+
+    fn add(&mut self, column: usize, v: impl Into<String>) {
+        self.objects.insert(column, v.into());
+        self.next_column = column + 1;
+    }
+
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>, end_column: usize) -> std::fmt::Result {
+        // Temporary string so we can trim whitespace at the end
+        let mut s = String::new();
+        match &self.wire {
+            Wire::Qubit { q_id: label } => {
+                s.write_str(&fmt_qubit_label(*label))?;
+                for column in 1..end_column {
+                    let val = self.objects.get(&column);
+                    if let Some(v) = val {
+                        s.write_str(v)?;
+                    } else {
+                        s.write_str(QUBIT_WIRE)?;
+                    }
+                }
+            }
+            Wire::Classical { start_column } => {
+                for column in 0..end_column {
+                    let val = self.objects.get(&column);
+                    if let Some(v) = val {
+                        s.write_str(v)?;
+                    } else if start_column.map_or(false, |s| column > s) {
+                        s.write_str(CLASSICAL_WIRE)?;
+                    } else {
+                        s.write_str(BLANK)?;
+                    }
+                }
+            }
+        }
+        writeln!(f, "{}", s.trim_end())?;
+        Ok(())
+    }
+}
+
+static COLUMN_WIDTH: usize = 7;
+static QUBIT_WIRE: &str = "───────";
+static CLASSICAL_WIRE: &str = "═══════";
+static QUBIT_WIRE_CROSS: &str = "───┼───";
+static CLASSICAL_WIRE_CROSS: &str = "═══╪═══";
+static CLASSICAL_WIRE_START: &str = "   ╘═══";
+static QUBIT_WIRE_DASHED_CROSS: &str = "───┆───";
+static CLASSICAL_WIRE_DASHED_CROSS: &str = "═══┆═══";
+static VERTICAL_DASHED: &str = "   ┆   ";
+static VERTICAL: &str = "   │   ";
+static BLANK: &str = "       ";
```

Comment:
All these can also be `const`. I think the difference is that if you mark it at `const` they get evaluated at compile time. So maybe you get some extra fancy optimizations from `rustc`.
```suggestion
const COLUMN_WIDTH: usize = 7;
const QUBIT_WIRE: &str = "───────";
const CLASSICAL_WIRE: &str = "═══════";
const QUBIT_WIRE_CROSS: &str = "───┼───";
const CLASSICAL_WIRE_CROSS: &str = "═══╪═══";
const CLASSICAL_WIRE_START: &str = "   ╘═══";
const QUBIT_WIRE_DASHED_CROSS: &str = "───┆───";
const CLASSICAL_WIRE_DASHED_CROSS: &str = "═══┆═══";
const VERTICAL_DASHED: &str = "   ┆   ";
const VERTICAL: &str = "   │   ";
const BLANK: &str = "       ";
```

Reply:
Makes sense, thanks!